ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V114__activate_string_interning_for_parties.sql

```diff
@@ -0,0 +1,317 @@
+  ALTER COLUMN flat_event_witnesses TYPE INTEGER[]
+  USING internalize_string_array('p|', flat_event_witnesses);
+
+ALTER TABLE participant_events_create
+  ALTER COLUMN create_signatories TYPE INTEGER[]
+  USING internalize_string_array('p|', create_signatories);
+
+ALTER TABLE participant_events_create
+  ALTER COLUMN create_observers TYPE INTEGER[]
+  USING internalize_string_array('p|', create_observers);
+
+
+
+SELECT 'Party interning data migration - Migrating participant_events_consuming_exercise...';
+
+DROP INDEX participant_events_consuming_exercise_flat_event_witnesses_idx;
+DROP INDEX participant_events_consuming_exercise_tree_event_witnesses_idx;
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN submitters TYPE INTEGER[]
+  USING internalize_string_array('p|', submitters);
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN tree_event_witnesses DROP DEFAULT;
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN tree_event_witnesses TYPE INTEGER[]
+  USING internalize_string_array('p|', tree_event_witnesses);
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN flat_event_witnesses DROP DEFAULT;
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN flat_event_witnesses TYPE INTEGER[]
+  USING internalize_string_array('p|', flat_event_witnesses);
+
+ALTER TABLE participant_events_consuming_exercise
+  ALTER COLUMN exercise_actors TYPE INTEGER[]
+  USING internalize_string_array('p|', exercise_actors);
+
+
+
+SELECT 'Party interning data migration - Migrating participant_events_non_consuming_exercise...';
+
+DROP INDEX participant_events_non_consuming_exercise_flat_event_witnes_idx;
+DROP INDEX participant_events_non_consuming_exercise_tree_event_witnes_idx;
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN submitters TYPE INTEGER[]
+  USING internalize_string_array('p|', submitters);
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN tree_event_witnesses DROP DEFAULT;
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN tree_event_witnesses TYPE INTEGER[]
+  USING internalize_string_array('p|', tree_event_witnesses);
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN flat_event_witnesses DROP DEFAULT;
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN flat_event_witnesses TYPE INTEGER[]
+  USING internalize_string_array('p|', flat_event_witnesses);
+
+ALTER TABLE participant_events_non_consuming_exercise
+  ALTER COLUMN exercise_actors TYPE INTEGER[]
+  USING internalize_string_array('p|', exercise_actors);
+
+
+
+SELECT 'Party interning data migration - Cleanup...';
+
+DROP INDEX string_interning_external_string_temp_idx;
+DROP FUNCTION internalize_string_array(TEXT, TEXT[]);
+
+CREATE VIEW participant_events
+AS
+SELECT
+    0::smallint as event_kind,
+    event_sequential_id,
+    NULL::text as event_offset,
```

Comment:
any reason to keep this one null?

Reply:
I don't know, but I think that is not part of this PR (I would prefer not to do unrelated changes).
To remove the view entirely is on the horizon, if we get there, I guess that would resolve this question as well.